mgo/find_test.go

```diff
@@ -139,16 +140,16 @@ func TestFind(t *testing.T) {
 				Limit:          2,
 				CountTotal:     true,
 			},
-			results: &[]item{},
+			results: &[]*item{},
```

Comment:
This is the only unit-test I modified to use a pointer. All of the others use values.
The main difference now is that we only rely on `results` being an array of some type rather than it being of the type `*[]item` specifically.